### PR TITLE
Feature/jwt auth login

### PR DESCRIPTION
**Pull request by ghelobytes**
# Description

Please include a summary of the change. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Instructions

Explain what someone needs to do in order to test the functionality of the changes. 

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: mrtt-ui/src/components/Auth/ProtectedRoutes.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  }
+
+  return (
+    <div>
+      <Outlet />
+    </div>
+  )
+}
+
+export default ProtectedRoutes
```
Comment (mmnoo) [inline mrtt-ui/src/components/Auth/ProtectedRoutes.js:17]: Neat. TIL about how to do this with react router 6 :)

File: mrtt-ui/src/views/Auth/LoginForm.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+const formDefaultValues = { email: '', password: '' }
+
+const LoginForm = () => {
+  const [isLoading] = useState(false)
+  const [isSubmitError, setIsSubmitError] = useState(false)
+  const [isSubmitting, setIsSubmitting] = useState(false)
+  const navigate = useNavigate()
+
+  const authUrl = `${getBaseUrl(process.env.REACT_APP_API_URL)}/users/sign_in`
```
Comment (mmnoo) [inline mrtt-ui/src/views/Auth/LoginForm.js:31]: Im thinking for future maintainability its probably better to make a new env var rather decrypting another url to glue something together. Maybe `REACT_APP_BASE_URL=https://mrtt-api-test-2.herokuapp.com/`. Then we can remove the `getBaseUrl` utility.   Explicit over implicit.

File: mrtt-ui/src/views/Auth/LoginForm.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  const signUp = (formData) => {
+    axios
+      .post(authUrl, { user: formData }, options)
+      .then(({ data }) => {
+        setIsSubmitting(false)
+        toast.success(data.message)
+
+        if (data.token) {
+          localStorage.setItem('token', data.token)
+          // TODO: navigate to another page
```
Comment (mmnoo) [inline mrtt-ui/src/views/Auth/LoginForm.js:55]: Lets avoid using the code base for project management. Can you make a ticket for this comment in our issue tracker instead?

**Review by mmnoo — CHANGES_REQUESTED** [review]
This looks fantastic! Easy to reason about code. Just two small requests.

File: mrtt-ui/src/views/Auth/LoginForm.js (hunk 2 above)
Comment (mmnoo) [reply mrtt-ui/src/views/Auth/LoginForm.js:31]: (sorry I didnt look at this helper function before to fully understand what it was doing)

File: mrtt-ui/src/views/Auth/LoginForm.js (hunk 3 above)
Comment (ghelobytes) [reply mrtt-ui/src/views/Auth/LoginForm.js:55]: Agree. I have another PR ready that addresses this.

File: mrtt-ui/src/views/Auth/LoginForm.js (hunk 2 above)
Comment (ghelobytes) [reply mrtt-ui/src/views/Auth/LoginForm.js:31]: Ok, will remove the helper function and use env var instead. Will use `REACT_APP_AUTH_URL` as it's purely related to auth.

**Comment by ghelobytes** [comment]
@mmnoo Change requests addressed by https://github.com/globalmangrovewatch/gmw-users/pull/151/commits/b9fdeec061159b7b0a42e4add818a1a89670e396

**Review by mmnoo — APPROVED** [review]
LGTM. If you werent already planning on it, can you squash merge this?

**Pull request merged**